grakel/utils.py

```diff
@@ -679,3 +688,108 @@ def get_el(ef, g):
+
+    A function for helping a user that has a bunch of graph csv
+    files make a collection for grakel input.
+
+    Parameters
+    ----------
+        data : Union[torch_geometric.data.Data, torch_geometric.data.batch.DataBatch]
+            Input should be a torch_geometric data object to be converted.
+
+        node_one_hot : bool, default=False
+            Whether each element of `data.x` should be treated as a one-hot vector.
+
+        edge_one_hot : bool, default=False
+            Whether each element of `data.edge_attr` should be treated as a one-hot vector.
+
+        ignore_y : bool, default=False
+            Argument determining whether to return the labels connected to the input graphs.
+
+    Returns
+    -------
+        grakel_graphs : dict
+            Returns a dictionary with keys:
+
+                - 'graph' : `list[grakel.Graph]`
+                    
+                    + If input is torch_geometric.data.Data returns a single element.
+                    
+                    + If input is torch_geometric.data.Data returns a list of graphs.
+
+                - 'y' : `list[int]`
+                    
+                    + List of labels of graphs.
+                    
+                    + If ignore_y is True this argument is ommited.
+
+    """
+    def one_hot_node(x):
+        if node_one_hot:
+            return x.argmax().item()
+        else:
+            return x.cpu().numpy()
+
+    def one_hot_edge(x):
+        if edge_one_hot:
+            return x.argmax().item()
+        else:
+            return x.cpu().numpy()
+
+    if data.batch is None:
+        edges = [tuple(e) for e in data.edge_index.t().tolist()]
+        node_labels, edge_labels = [], []
+        if data.x is not None:
+            node_labels = {i: one_hot_node(data.x[i]) for i in range(data.x.shape[0])}
+
+        if data.edge_attr is not None:
+            edge_labels = {edges[i]: one_hot_edge(data.edge_attr[i]) for i in range(data.edge_attr.shape[0])}
+
+        x = {'graph': Graph(edges, node_labels, edge_labels)}
+        if data.y is not None:
+            try:
+                x['y'] = int(data.y.item())
+            except:
+                raise ValueError('Either set ignore_y=True or use a ')
+
+        return x
+    else:
+        edges = defaultdict(list)
+        node_labels = None
+        edge_labels = None if data.edge_attr is None else defaultdict(dict)
+        group_id = (data.batch).unique_consecutive().tolist()
+        lookup = {i: data.batch[i].item() for i in range(data.batch.shape[0])}
+        for i, e in enumerate(data.edge_index.t().tolist()):
+            e = tuple(e)
+            group = lookup[e[0]]
+            if lookup[e[0]] != lookup[e[1]]:
+                raise ValueError('Both nodes from an edge should correspond to the same graph.')
+            edges[group].append(e)
+            if  data.edge_attr is not None:
+                edge_labels[group][e] = one_hot_edge(data.edge_attr[i])
```

Comment:
This should be replaced with:
```
edges = defaultdict(list)
node_labels = None
edge_labels = None if data.edge_attr is None else defaultdict(dict)
group_id, counts = (data.batch).unique_consecutive(return_counts=True)
group_id = group_id.tolist()
cumsum_n_nodes = counts.cumsum(0).tolist()
cumsum_n_nodes.insert(0, 0)
lookup = {i: data.batch[i].item() for i in range(data.batch.shape[0])}
for i, e in enumerate(data.edge_index.t().tolist()):
    group = lookup[e[0]]
    if lookup[e[0]] != lookup[e[1]]:
        raise ValueError('Both nodes from an edge should correspond to the same graph.')
    e = (e[0]-cumsum_n_nodes[group], e[1]-cumsum_n_nodes[group])
    edges[group].append(e)
    if  data.edge_attr is not None:
        edge_labels[group][e] = one_hot_edge(data.edge_attr[i])
```

Reply:
it was a bug of the __repr__
- do we really need to remap them?
Imagine a user wants to retrieve them by name.

Reply:
@giannisnik

Reply:
If the key of the dictionary is just the name of the node, and there is already a mapping from node names to node indices, you are right, we do not need to map them.

Reply:
yes it was just a printing problem of the __repr__ you can try again :))